Add support for full screen sharing under iOS

jcamins opened the pull request:
This PR adds support for sharing the user's full screen under iOS using ReplayKit (with an appropriate app extension). There are still changes that need to be made in the xcode project for the app in order to enable it (documented in `docs/IOS-SCREENSHARE.md`), but these changes allow the library to be used to enable screen sharing without any further modification of the library code.

jcamins commented on ios/RNZoomUs.m:
```diff
@@ -239,4 +243,24 @@ - (void)onMeetingError:(MobileRTCMeetError)errorCode message:(NSString *)message
 }
 
+- (void)onClickShareScreen:(UIViewController *)parentVC {
+  if (@available(iOS 12.0, *)) {
```
This section prefills the iOS SystemBroadcastPicker (accessible from the button in the Control Center) with the extension the developer has implemented as part of the same xcode project.

jcamins commented on ios/RNZoomUs.m:
```diff
@@ -239,4 +243,24 @@ - (void)onMeetingError:(MobileRTCMeetError)errorCode message:(NSString *)message
+
+- (BOOL)respondsToSelector:(SEL)aSelector
+{
+    if (aSelector == @selector(onClickShareScreen:)) {
```
If no `screenShareExtension` has been configured, we will report to Zoom that our `MobileRTCMeetingServiceDelegate` doesn't support `onClickShareScreen:`.

mieszko4 commented on docs/IOS-SCREENSHARE.md:
```diff
@@ -0,0 +1,22 @@
+await ZoomUs.initialize({
+  clientKey: '...',
+  cientSecret: '...',
+  appGroupId: 'group.com.your.Bundle',
+  screenShareExtension: 'com.your.Bundle.ScreenShare'
```
1. since these are only ios-related I think it would be good to prefix them, something like:
```suggestion
  iosAppGroupId: 'group.com.your.Bundle',
  iosScreenShareExtensionId: 'com.your.Bundle.ScreenShare'
```
2. also note how I added `Id` in the second param

mieszko4 reviewed (approved):
looking good thnx.
If I understood correctly, adding `MobileRTCScreenShare.framework` is still optional if we do not want to integrate screen-share, right?

jcamins commented:
Updated to prefix the two parameters. And to confirm you're understanding, if you don't add the `MobileRTCScreenShare` framework, you won't be able to share your screen, but the meeting will still work fine (if you specify an `iosScreenShareExtension` in that case, it'll show a "Screen" option under "Share Content" that doesn't work, but I can't come up with any way to avoid this).

mieszko4 commented:
@jcamins great, thnx! Jus one more note regarding naming: you use `iosAppGroupId` and `iosScreenShareExtension`.
We should use `iosScreenShareExtensionId` for consistency, no?

jcamins commented:
Whoops, sorry. You even pointed that out before. Updated.

By the way, it looks like you're only one who can merge PRs into this repo, so I'm going to assume now that I've addressed the comments these are good to go. Let me know if you need anything else!

mieszko4 commented:
Oh, ok, I thought that once I approve you can merge. Anyway I will merge it now, thnx!
